src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodosCategory } from './types/TodosCategory';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState<TodosCategory | string>(TodosCategory.All);
+
+  const filterTodos = () => {
+    const filteredTodos = todos.filter(todo => {
+      const includesQuery = todo.title.toLowerCase()
+        .includes(query.toLowerCase());
+
+      switch (filterType) {
+        case TodosCategory.Active:
+          return !todo.completed && includesQuery;
+        case TodosCategory.Completed:
+          return todo.completed && includesQuery;
+        default:
+          return includesQuery;
+      }
+    });
+
+    return filteredTodos;
+  };
+
+  useEffect(() => {
+    const getTodosFromApi = async () => {
+      try {
+        const response = await getTodos();
+
+        setTodos(response);
+        setIsLoaded(true);
+      } catch (error) {
+        throw new Error('Error on loading todos');
+      }
```

Comment:
What's the point in throwing error if error is already thrown?

Let's show message on UI instead

Reply:
Denis said that we need do it like that, so i think it`s not critical